### PR TITLE
Ddcnl 3973

### PR DESCRIPTION
**Pull request by oliwiaKot**
- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: app/uk/gov/hmrc/tai/forms/income/previousYears/UpdateIncomeDetailsForm.scala [hunk 1]
```diff
@@ -47,8 +47,8 @@ object UpdateIncomeDetailsForm {
 
   def textExceedsCharacterLimit(exceedErrorMsg: String): Constraint[String] =
     Constraint[String]("characterLimitExceeded") {
-      case textValue if textValue.trim.length <= historicEmploymentDetailsCharLimit => Valid
-      case _                                                                        => Invalid(exceedErrorMsg)
+      case textValue if textValue.trim.replace("\r", "").length <= historicEmploymentDetailsCharLimit => Valid
```
Comment (struong) [inline app/uk/gov/hmrc/tai/forms/income/previousYears/UpdateIncomeDetailsForm.scala:50]: Missing a test, also I believe this is the same as doing `stripLineEnd` ?

File: app/controllers/income/previousYears/UpdateIncomeDetailsController.scala [hunk 2]
```diff
@@ -109,7 +109,7 @@ class UpdateIncomeDetailsController @Inject()(
       },
       incomeDetails => {
         journeyCacheService
-          .cache(Map(UpdatePreviousYearsIncome_IncomeDetailsKey -> incomeDetails))
+          .cache(Map(UpdatePreviousYearsIncome_IncomeDetailsKey -> incomeDetails.replace("\r", "")))
```
Comment (struong) [inline app/controllers/income/previousYears/UpdateIncomeDetailsController.scala:112]: Do we need this bit? The validation should have stripped it by the time it comes to caching?

Comment (oliwiaKot) [reply app/controllers/income/previousYears/UpdateIncomeDetailsController.scala:112]: I checked and it doesn't, after the validation there's still the additional character present when we come to the controller

File: app/uk/gov/hmrc/tai/forms/income/previousYears/UpdateIncomeDetailsForm.scala (hunk 1 above)
Comment (oliwiaKot) [reply app/uk/gov/hmrc/tai/forms/income/previousYears/UpdateIncomeDetailsForm.scala:50]: stripLineEnd doesn't work of this, it only strips a trailing new line at the end of the whole string, it wouldn't handle new lines in the middle of the text

Comment (oliwiaKot) [reply app/uk/gov/hmrc/tai/forms/income/previousYears/UpdateIncomeDetailsForm.scala:50]: spec added

**Review by struong — APPROVED** [review]

**Pull request merged**